partiql-eval/src/eval.rs

```diff
@@ -588,6 +588,22 @@ impl EvalExpr for EvalBinOpExpr {
     }
 }
 
+#[derive(Debug)]
+pub struct EvalBetweenExpr {
+    pub value: Box<dyn EvalExpr>,
+    pub from: Box<dyn EvalExpr>,
+    pub to: Box<dyn EvalExpr>,
+}
+
+impl EvalExpr for EvalBetweenExpr {
+    fn evaluate(&self, bindings: &Tuple, ctx: &dyn EvalContext) -> Value {
+        let value = self.value.evaluate(bindings, ctx);
+        let from = self.from.evaluate(bindings, ctx);
+        let to = self.to.evaluate(bindings, ctx);
+        value.gteq(&from).and(&value.lteq(&to))
```

Comment:
Previously, these `Value` methods (`gteq` and `lteq` and `and`) used `Value`s as parameters, which took ownership of the `Value`s. Changing the methods to use references allowed me to reuse the `Value`s without cloning. Not sure if this was the best way to reuse the `Value`s.